### DISCUSSION TITLE
Welcome to ArcWelderLib Discussions!

### DISCUSSION THREAD

**Discussion by FormerLurker** [FormerLurker/ArcWelderLib#35] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by richg99** [comment]
Just a question from a newbie to Formerlurker.  re Arc Welder. I loaded it and tried it today on Cura for my Ender 3V2. It seemed to add tons of time to an expected print. If I read some of the comments correctly, ...since I am using the SD card at all times, Arc Welder won't help me? Is that correct?  I'm new so I probably don't understand all that I should about this. The concept sounds amazing. Thank You.

**Reply by FormerLurker** [reply]
I'd like to see your gcode if possible.  Can you post the before and after gcode to gist.github.com and drop a link in here?

> since I am using the SD card at all times, Arc Welder won't help me? Is that correct?

Well, it might.  It's really difficult to tell without knowing a lot more about your firmware, slicer settings, etc...  Still, I'd like to figure out why the welded gcode is so slow.  It should, in the worst case, have no effect.

What firmware are you using (a link to the source would be great) and what are your arc settings in your firmware?

**Comment by richg99** [comment]
Ha Ha    I am so new that I don't know how to do any of those things.
I deleted the Cura Slice after it showed something like Two days on a slice
that, previous to adding Arc Welder, was only 6 hours. I did finally print
the item at about 6 hours, after I deleted the Arc Welder option.

I'll try to recreate, and save the code that I didn't use for you. This
will take this old 81 yr old guy some time, but I'll do it.

regards, rich

**Reply by FormerLurker** [reply]
Rich, that is awesome!  Typically I'm the senior member of these discussions, so this is unusual for me (in my 40s), but the diversity warms my soul.   You sound like you are learning very fast if you are new and your diving into bleeding edge 3d printer topics like G2/G3!  Also, posting issues on Github too!  I'm wondering what other cutting edge tech you are into, and what brought you into the hobby.  Shoot me an email (formerlurker@pm.me) and I'll invite you to the Octolapse discord server (haven't created one for ArcWelder), and you can meet some of the gang and we can chat.

> Cura Slice after it showed something like Two days on a slice
that, previous to adding Arc Welder, was only 6 hours.

So this was just a print time estimation problem, as in the actual welded print finished in six hours?  If so, this is what we call an upstream issue (the parent program, or cura in this case).  Cura doesn't generate or visualize arcs, so it's not totally unexpected that it would have problems here.  Now if your print took two days, but only should have taken six hours, that is a HUGE issue.

**Comment by richg99** [comment]
Well, unfortunately, I deleted the actual .OBJ file that showed the issues.
I tried to recreate the problem using a later file, but, as you said,
whether I had ArcWelder ON, or OFF, the estimated time was exactly the
same. I will keep messing around, and if I can pop up the same results,
I'll get the info to you.

If I read things correctly, though, the major advantage of Arc Welder will
be for someone who uses Octi or somehow uses a USB input. Those Luddites,
like myself, who prefer just sticking a card in, shouldn't derive much
benefit???

As far as whether the issues were in Cura, or in my Ender, as opposed to
being actually an issue with your program, I'd vote for the issues NOT
being in ArcWelder. My project ( a fishing lure mold) was projected on Cura
as a 6-hour print. On my Ender3-V2, it showed 60+ HOURS!.  Fifteen minutes
into the print it showed 38 hours remaining, yet produced a good print in
6+hours!

Yes, I do get myself into some strange things given my age. I did my first,
( and only) acting job at age 75 (Chief Brompton in Who Flew Over the
Cuckoo's Nest); took up Skeet and Trapshooting at age 80, and bought my
first 3D printer at age 81. I have no business being on GitHub, but it
seemed to be the only way that I could send a message to you about your new
program!

As life rolls on, just remember that you aren't dead until you think you
are!
rich gross   richg99@gmail.com

p.s. Also, I just joined e-Nable. It is a worldwide organization that 3D
prints, for free, artificial limbs for indignant youngsters who have lost
theirs.

**Reply by FormerLurker** [reply]
> On my Ender3-V2, it showed 60+ HOURS!.  Fifteen minutes
into the print it showed 38 hours remaining, yet produced a good print in
6+hours!

Was this printed via SD or within Octoprint or something?

**Comment by richg99** [comment]
I only use an SD card at this time.

**Reply by FormerLurker** [reply]
Ok, it's a firmware estimation then.  This is a downstream problem that might not be fixable.  However... I think some slicers can insert gcode showing a more accurate estimation...  Hmm..  will think about this.

**Comment by richg99** [comment]
Please note. The most current issue came up without Arc welder being
activated. Your program wasn't the problem.

**Comment by JsemOndra** [comment]
Hi FormerLurker,
first of all thank you for this neat tool! It really improved my 3D printer expirience!
I was wondering about one thing I havent found in the docs. Is there a way how to define minimum radius or arcs to be converted? For cases when I dont want to convert radiuses in infill and I'd like to set for example 3mm minimum radius of arc to be processed?

Thank you,
35mmHunter

**Reply by FormerLurker** [reply]
Well, there sort of is, but not exactly what you want.  It would be possible to use 'firmware compensation' for this, but you'd have to calculate the proper parameters.

I'm unsure how this is helpful exactly, though.  Why do you not want some areas to be converted exactly?  Perhaps with more info the original issue you are having can be remedied.

**Reply by JsemOndra** [reply]
I'll look into firmware compensation option, thank you :)
The reason I'll need is because with G2/G3 the printing time of same part increases (e.g. in my case from 9hrs to 11hrs) and if I could specify limit for arc radius, then infill would still be printed using G1 movements (=faster) and just faces using G2/G3.

**Comment by FormerLurker** [comment]
@35mmHunter, 

> The reason I'll need is because with G2/G3 the printing time of same part increases (e.g. in my case from 9hrs to 11hrs)

Have you verified this by printing and timing the part?  If so, does it actually print slower?  If so, this sounds like a firmware issue because arc welder should never increase print time.  In fact, it can even reduce it slightly if there is any planner starvation (especially when streaming over USB).

**Reply by JsemOndra** [reply]
Yes, I actually found out by printing the part. I used very same g-code as I am using normally, run it through the ArcWelder and printed. And on Marlin 2.0.x (I'll check the exact fw version later today) running on Anet A8 original board (with custom aluminum x-frame) it increased the print time.

**Reply by FormerLurker** [reply]
Well, that is something different.  I've heard of this before, but only when using certain firmware.  Maybe you can look at the arc interpolation settings and paste them in.  I'd hate to add a workaround to what is either a but in ArcWelder or the Firmware when the original problem could be fixed.

**Reply by JsemOndra** [reply]
Agree! I'll paste the part as soon as I'll get home and find it. Thanks for all the help :)

**Comment by andraisr** [comment]
Greetings,
I came across ArcWelder during my search for improving the surface of my prints. I have been 3D printing for 3 years and am very verse on getting good prints, just looking for better prints. I did encounter an issue that I documented with my install.  
I am not seeing the post processing when I view the gcode file. I do see where it used the process I created from S3D. I entered the path correctly, so I know that is correct. I copy/paste the path  without the exe name into my browser and it took me to the correct folder. So I know I did not add any hidden characters. This is the text I put into the Post Processing area in S3D ....   D:\aw\Windows_1.1.1\bin\ArcWelder.exe "[output_filepath]"

I also attempted to run the program from the command line. I do see it processing and get info on the screen. But there is not an output gcode file created.

Your thoughts would be greatly appreciated. Thanks in advance!

**Reply by FormerLurker** [reply]
If you are only supplying one parameter, the "[output_filepath]" in your S3D parameters, the source file will be overwritten.  Did you try opening the file and looking at the top for a header added by arcwelder?  You can also search the file for G2/G3.

**Reply by andraisr** [reply]
Thanks for responding. Yes, I did look at the source file to review the header info. The header still refers to S3D. Searched the file for G2/G3 command. Did not find any, everything was still G1. The object is a curvy boomerang, so I know there should be at least one G2 command in it. If I wanted to ooutput the file to a new file name from ArcWelder, using the example of how I have it on the previous msg, what would the text look like...  I am currently using ...   D:\aw\Windows_1.1.1\bin\ArcWelder.exe "[output_filepath]" in the post processing section of S3D.

**Reply by andraisr** [reply]
More info... I can get it to work thru the command line. Not working thru S3d. I also determined that it does not work with the Geeetech A30T printer. This print is not Marlin firmware based. It does not recognize the G2/G3 commands. It just stops and sits forever when it encounters those commands.I can print with my A10T ( this one is Marlin firmware based). However a 46 minute print takes 3 hours with ArcWelder. When printing curves, it runs nice and fast, but switching back to the G1 commands it dogs down. I print the same stl from S3D and it runs much faster. My A10T has TMC2208 and tl-smoothie boards (full bridge rectifiers on each of the 4 wires going to the stepper motors). and the output this combination and ArcWelder is amazing. But Slooowwwww.

**Reply by FormerLurker** [reply]
@andraisr,

> I also determined that it does not work with the Geeetech A30T printer.

You may consider seeing if you can run marlin on this printer.  I personally am not a fan of overly customized firmware, but YMMV.

> However a 46 minute print takes 3 hours with ArcWelder

Have you printed the object without arc welder to verify?  It could be an estimation issue with g2/g3.  If not, I'd need to see the firmware's config file and base version to figure out what's going on.  Arc support has been greatly enhanced in recent versions of marlin, and a ton of the custom forks I see are running really old code, and a lot of them are misconfigured.

**Reply by andraisr** [reply]
Unfortunately,the A30T only works with Geetech's Smarttool firmware. There have been attempts to convert it to Marlin but the A30T uses a color touch screen and no one has been able to make that work with marlin.

I did print the gcode file created by ArcWelder and the original file the I used the file from ArcWelder. 

I am currently printing a funnel gcode, then will reprint using the file output from Acrwelder, just to see if it behaves differently regarding print time.

I am using Marlin 2.0.1 on the A10T. I have arc support enabled. I have been attempting to figure out why 2.0.9.1 will not produce an output file ( i know it has to do with the workspace params, not versed enough in Platformio to get it going, yet). Here are the config and adv config I am using for the A10T. 

[Marlin.zip](https://github.com/FormerLurker/ArcWelderLib/files/6831082/Marlin.zip)

Anything you see off, please let me know.

**Comment by andraisr** [comment]
Any updates on my marlin??

**Reply by FormerLurker** [reply]
Everything looks OK, so I suspect some issues with the Junction Deviation algorithm in 2.0.1.  These were resolved in a later version, so I'd recommend spending a bit more time with 2.0.9.1.  It may take you a while, but it will be worth it!
